subprojects/core/src/main/java/org/gradle/execution/plan/FinalizerGroup.java

```diff
@@ -260,20 +257,33 @@ private MemberSuccessors getNodesThatBlock(Node node) {
         return successors.getNodesThatBlock(node);
     }
 
-    private boolean dependsOn(Node fromNode, Node toNode) {
-        Set<Node> seen = new HashSet<>();
-        List<Node> queue = new ArrayList<>();
-        Iterables.addAll(queue, fromNode.getHardSuccessors());
-        while (!queue.isEmpty()) {
-            Node node = queue.remove(0);
-            if (node == toNode) {
-                return true;
-            }
-            if (!seen.add(node)) {
-                continue;
+    private Set<Node> getDependenciesThatAreMembers(Set<Node> blockedFinalizedMembers) {
+        Set<Node> dependenciesThatAreMembers = new HashSet<>(members.size());
+        Set<Node> seen = new HashSet<>(1024);
```

Comment:
Only in the sense that it isn't the default.  We have no way of estimating what these numbers would actually be, but since the growth algorithm doubles the set size when growing, starting with a larger number means we grow more quickly for large graphs.  This was just a swag at what might be a reasonable starting point without being too large for small graphs.  TBH, this only had a slight impact on performance, but it was noticeable.